libmachine/provision/redhat.go

```diff
@@ -114,12 +114,31 @@ func (provisioner *RedHatProvisioner) dockerDaemonResponding() bool {
 	return true
 }
 
+func (provisioner *RedHatProvisioner) disableNetworkManagerSetupService() error {
+	if _, err := provisioner.SSHCommand("sudo systemctl disable nm-cloud-setup.service nm-cloud-setup.timer"); err != nil {
+		return err
+	}
+
+	// ignore errors here because the SSH connection will close
+	provisioner.SSHCommand("sudo reboot")
+
+	log.Debug("NetworkManager patched, waiting for machine to reboot...")
+	return drivers.WaitForSSH(provisioner.Driver)
+}
+
 func (provisioner *RedHatProvisioner) Provision(swarmOptions swarm.Options, authOptions auth.Options, engineOptions engine.Options) error {
 	provisioner.SwarmOptions = swarmOptions
 	provisioner.AuthOptions = authOptions
 	provisioner.EngineOptions = engineOptions
 	swarmOptions.Env = engineOptions.Env
 
+	if provisioner.OsReleaseInfo.VersionID == "8.4" {
```

Comment:
I like the idea of putting this check inside the function and changing the name to `disableNetworkManagerSetupService8dot4`